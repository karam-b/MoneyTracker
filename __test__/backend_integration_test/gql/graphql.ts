/* eslint-disable */
export type Maybe<T> = T | null
export type InputMaybe<T> = Maybe<T>
export type Exact<T extends { [key: string]: unknown }> = {
  [K in keyof T]: T[K]
}
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & {
  [SubKey in K]?: Maybe<T[SubKey]>
}
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & {
  [SubKey in K]: Maybe<T[SubKey]>
}
export type MakeEmpty<
  T extends { [key: string]: unknown },
  K extends keyof T,
> = { [_ in K]?: never }
export type Incremental<T> =
  | T
  | {
      [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never
    }
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: { input: string; output: string }
  String: { input: string; output: string }
  Boolean: { input: boolean; output: boolean }
  Int: { input: number; output: number }
  Float: { input: number; output: number }
  Date: { input: any; output: any }
}

export type Category = {
  __typename?: 'Category'
  color?: Maybe<Scalars['String']['output']>
  createdAt: Scalars['Date']['output']
  createdBy: User
  icon?: Maybe<Scalars['String']['output']>
  id: Scalars['ID']['output']
  title: Scalars['String']['output']
  updatedAt: Scalars['Date']['output']
}

export type CategoryInput = {
  color?: InputMaybe<Scalars['String']['input']>
  icon?: InputMaybe<Scalars['String']['input']>
  title: Scalars['String']['input']
}

export type Entry = {
  __typename?: 'Entry'
  amount: Scalars['Float']['output']
  category?: Maybe<Category>
  createdAt: Scalars['Date']['output']
  createdBy: User
  id: Scalars['ID']['output']
  note?: Maybe<Scalars['String']['output']>
  title: Scalars['String']['output']
  updatedAt: Scalars['Date']['output']
}

export type EntryInput = {
  amount: Scalars['Float']['input']
  category?: InputMaybe<Scalars['ID']['input']>
  note?: InputMaybe<Scalars['String']['input']>
  title: Scalars['String']['input']
}

export type Mutation = {
  __typename?: 'Mutation'
  createManyCategories: Array<Category>
  createManyEntries: Array<Entry>
  createOneCategory: Category
  createOneEntry: Entry
  deleteManyCategories: Scalars['Int']['output']
  deleteManyEntries: Scalars['Int']['output']
  deleteOneCategory: Scalars['Boolean']['output']
  deleteOneEntry: Scalars['Boolean']['output']
  updateCurrentUser: User
  updateOneCategory: Scalars['Boolean']['output']
  updateOneEntry: Scalars['Boolean']['output']
  updatePassword: Scalars['Boolean']['output']
}

export type MutationCreateManyCategoriesArgs = {
  categories: Array<CategoryInput>
}

export type MutationCreateManyEntriesArgs = {
  entries: Array<EntryInput>
}

export type MutationCreateOneCategoryArgs = {
  category: CategoryInput
}

export type MutationCreateOneEntryArgs = {
  entry: EntryInput
}

export type MutationDeleteManyCategoriesArgs = {
  ids: Array<Scalars['ID']['input']>
}

export type MutationDeleteManyEntriesArgs = {
  ids: Array<Scalars['ID']['input']>
}

export type MutationDeleteOneCategoryArgs = {
  id: Scalars['ID']['input']
}

export type MutationDeleteOneEntryArgs = {
  id: Scalars['ID']['input']
}

export type MutationUpdateCurrentUserArgs = {
  user: UserInput
}

export type MutationUpdateOneCategoryArgs = {
  category: CategoryInput
  id: Scalars['ID']['input']
}

export type MutationUpdateOneEntryArgs = {
  entry: EntryInput
  id: Scalars['ID']['input']
}

export type MutationUpdatePasswordArgs = {
  password: Scalars['String']['input']
}

export type PaginationRequest = {
  page: Scalars['Int']['input']
  pageSize: Scalars['Int']['input']
}

export type PaginationResponse = {
  __typename?: 'PaginationResponse'
  data: Array<Entry>
  page: Scalars['Int']['output']
  pageSize: Scalars['Int']['output']
  total: Scalars['Int']['output']
  totalPages: Scalars['Int']['output']
}

export type Query = {
  __typename?: 'Query'
  getAllCategories: Array<Category>
  getAllEntries: PaginationResponse
  getCurrentUser: User
  getOneCategory?: Maybe<Category>
  getOneEntry?: Maybe<Entry>
}

export type QueryGetAllEntriesArgs = {
  pagination: PaginationRequest
}

export type QueryGetOneCategoryArgs = {
  id: Scalars['ID']['input']
}

export type QueryGetOneEntryArgs = {
  id: Scalars['ID']['input']
}

export type User = {
  __typename?: 'User'
  avatar?: Maybe<Scalars['String']['output']>
  createdAt: Scalars['Date']['output']
  displayName?: Maybe<Scalars['String']['output']>
  email: Scalars['String']['output']
  id: Scalars['ID']['output']
  providers: Scalars['String']['output']
  updatedAt: Scalars['Date']['output']
}

export type UserInput = {
  avatar?: InputMaybe<Scalars['String']['input']>
  displayName?: InputMaybe<Scalars['String']['input']>
}
