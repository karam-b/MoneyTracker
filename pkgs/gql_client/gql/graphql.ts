/* eslint-disable */
import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
export type MakeEmpty<T extends { [key: string]: unknown }, K extends keyof T> = { [_ in K]?: never };
export type Incremental<T> = T | { [P in keyof T]?: P extends ' $fragmentName' | '__typename' ? T[P] : never };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: { input: string; output: string; }
  String: { input: string; output: string; }
  Boolean: { input: boolean; output: boolean; }
  Int: { input: number; output: number; }
  Float: { input: number; output: number; }
  Date: { input: any; output: any; }
};

export type Category = {
  __typename?: 'Category';
  color?: Maybe<Scalars['String']['output']>;
  createdAt: Scalars['Date']['output'];
  createdBy: User;
  icon?: Maybe<Scalars['String']['output']>;
  id: Scalars['ID']['output'];
  title: Scalars['String']['output'];
  updatedAt: Scalars['Date']['output'];
};

export type CategoryInput = {
  color?: InputMaybe<Scalars['String']['input']>;
  icon?: InputMaybe<Scalars['String']['input']>;
  title: Scalars['String']['input'];
};

export type Entry = {
  __typename?: 'Entry';
  amount: Scalars['Float']['output'];
  category: Category;
  createdAt: Scalars['Date']['output'];
  createdBy: User;
  id: Scalars['ID']['output'];
  note?: Maybe<Scalars['String']['output']>;
  title: Scalars['String']['output'];
  updatedAt: Scalars['Date']['output'];
};

export type EntryInput = {
  amount?: InputMaybe<Scalars['Float']['input']>;
  category?: InputMaybe<Scalars['ID']['input']>;
  note?: InputMaybe<Scalars['String']['input']>;
  title: Scalars['String']['input'];
};

export type Mutation = {
  __typename?: 'Mutation';
  createManyCategories: Array<Category>;
  createManyEntries: Array<Entry>;
  createOneCategory: Category;
  createOneEntry: Entry;
  deleteManyCategories: Scalars['Int']['output'];
  deleteManyEntries: Scalars['Int']['output'];
  deleteOneCategory: Scalars['Boolean']['output'];
  deleteOneEntry: Scalars['Boolean']['output'];
  updateCurrentUser: User;
  updateOneCategory: Scalars['Boolean']['output'];
  updateOneEntry: Scalars['Boolean']['output'];
  updatePassword: Scalars['Boolean']['output'];
};


export type MutationCreateManyCategoriesArgs = {
  categories: Array<CategoryInput>;
};


export type MutationCreateManyEntriesArgs = {
  entries: Array<EntryInput>;
};


export type MutationCreateOneCategoryArgs = {
  category: CategoryInput;
};


export type MutationCreateOneEntryArgs = {
  entry: EntryInput;
};


export type MutationDeleteManyCategoriesArgs = {
  ids: Array<Scalars['ID']['input']>;
};


export type MutationDeleteManyEntriesArgs = {
  ids: Array<Scalars['ID']['input']>;
};


export type MutationDeleteOneCategoryArgs = {
  id: Scalars['ID']['input'];
};


export type MutationDeleteOneEntryArgs = {
  id: Scalars['ID']['input'];
};


export type MutationUpdateCurrentUserArgs = {
  user: UserInput;
};


export type MutationUpdateOneCategoryArgs = {
  category: CategoryInput;
  id: Scalars['ID']['input'];
};


export type MutationUpdateOneEntryArgs = {
  entry: EntryInput;
  id: Scalars['ID']['input'];
};


export type MutationUpdatePasswordArgs = {
  password: Scalars['String']['input'];
};

export type Query = {
  __typename?: 'Query';
  getAllCategories: Array<Category>;
  getAllEntries: Array<Entry>;
  getCurrentUser: User;
  getOneCategory?: Maybe<Category>;
  getOneEntry?: Maybe<Entry>;
};


export type QueryGetOneCategoryArgs = {
  id: Scalars['ID']['input'];
};


export type QueryGetOneEntryArgs = {
  id: Scalars['ID']['input'];
};

export type User = {
  __typename?: 'User';
  avatar?: Maybe<Scalars['String']['output']>;
  createdAt: Scalars['Date']['output'];
  displayName?: Maybe<Scalars['String']['output']>;
  email: Scalars['String']['output'];
  id: Scalars['ID']['output'];
  providers: Scalars['String']['output'];
  updatedAt: Scalars['Date']['output'];
};

export type UserInput = {
  avatar?: InputMaybe<Scalars['String']['input']>;
  displayName?: InputMaybe<Scalars['String']['input']>;
};

export type Get_Current_ProfileQueryVariables = Exact<{ [key: string]: never; }>;


export type Get_Current_ProfileQuery = { __typename?: 'Query', getCurrentUser: { __typename?: 'User', id: string, email: string, displayName?: string | null, avatar?: string | null, providers: string, createdAt: any, updatedAt: any } };


export const Get_Current_ProfileDocument = {"kind":"Document","definitions":[{"kind":"OperationDefinition","operation":"query","name":{"kind":"Name","value":"get_current_profile"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"getCurrentUser"},"selectionSet":{"kind":"SelectionSet","selections":[{"kind":"Field","name":{"kind":"Name","value":"id"}},{"kind":"Field","name":{"kind":"Name","value":"email"}},{"kind":"Field","name":{"kind":"Name","value":"displayName"}},{"kind":"Field","name":{"kind":"Name","value":"avatar"}},{"kind":"Field","name":{"kind":"Name","value":"providers"}},{"kind":"Field","name":{"kind":"Name","value":"createdAt"}},{"kind":"Field","name":{"kind":"Name","value":"updatedAt"}}]}}]}}]} as unknown as DocumentNode<Get_Current_ProfileQuery, Get_Current_ProfileQueryVariables>;