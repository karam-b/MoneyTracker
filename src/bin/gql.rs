use async_graphql::*;

struct Query;

#[Object]
impl Query {
    async fn hello(&self, name: String) -> String {
        format!("hello {name}")
    }
}

#[tokio::main]
async fn main() -> Result<()> {
    let schema = Schema::new(Query, EmptyMutation, EmptySubscription);

    let query = "{ hello(name: \"karam\") }";

    let res = schema.execute(query).await;

    println!("{}", res.data.into_json().unwrap());

    Ok(())
}
