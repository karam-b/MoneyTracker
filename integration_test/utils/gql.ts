import { graphql } from "../gql/gql";

export default function gql(literals) {
  const rt = graphql(literals[0]);

  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  rt.__as_str = literals[0];
  return graphql(literals[0]);
}

export function getAsString(rt: ReturnType<typeof gql>) {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  return rt.__as_str;
}
