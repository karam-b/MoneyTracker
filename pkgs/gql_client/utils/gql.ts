import { graphql } from "../gql/gql";

export default function gql(literals, ...placeholders) {
  // const input = literals.join("");
  const input = literals.reduce((acc, cur, idx) => {
    return acc + cur + (placeholders[idx] || "");
  }, "");

  const rt = graphql(input);

  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  rt.__my_input = input;
  return graphql(input);
}

export function getGqlInput(rt: ReturnType<typeof gql>) {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  return rt.__my_input;
}
