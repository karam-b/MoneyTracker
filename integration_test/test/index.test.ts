import { it, expect } from "@jest/globals";
import gql, { getAsString } from "../utils/gql";

const posts = gql`
  query new {
    getAllCategories {
      id
    }
  }
`;

const obj = {} as any;

it("test1", async () => {
  expect(getAsString(posts)).toEqual(true);
});
