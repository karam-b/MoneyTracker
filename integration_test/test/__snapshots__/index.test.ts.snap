// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root 1`] = `
Object {
  "doc": "",
}
`;

exports[`test1 1`] = `Promise {}`;
